v3.2-core/gl/procaddr.go

```diff
@@ -14,8 +14,9 @@ package gl
 #cgo windows LDFLAGS: -lopengl32
 #cgo darwin CFLAGS: -DTAG_DARWIN
 #cgo darwin LDFLAGS: -framework OpenGL
-#cgo linux freebsd CFLAGS: -DTAG_POSIX
+#cgo linux freebsd CFLAGS: -DTAG_POSIX -I/usr/local/include
```

Comment:
I'm not a FreeBSD expert, but as I understand it, shouldn't the `/usr/local/include` include file path be configured as part of the C compiler configuration? In other words, setting it here (in the Go package source code) is the wrong place. Am I missing something? /cc @errcw

Reply:
Yeah I had the same thought belatedly after approving. I also thought that this change should land upstream in glow first.

Reply:
Yes, it should land in `glow` first. We can review it here, and once it's ready, update `glow` and regenerate.

But, I'm not sure that this is the right fix.

I also don't know if the `pkg-config: gl` line is needed or not. How did it work on FreeBSD for other people (who added the FreeBSD support in the first place)?

Reply:
The `-I` addition shouldn't be made here; it's specific to some systems, and made redundant by pkg-config. Using `pkg-config`, however, is (theoretically) the correct way to add all the required libraries, include paths, cflags and so on needed to link against `gl`. I reckon some Linux distributions, such as nixos, also require it.

The way the patch is written, however, it is slightly incorrect, because you want to be able to choose between GL and EGL. A more appropriate patch would look something like this (untested):

```
--- procaddr.go.orig	2018-04-08 19:25:59.512241000 +0200
+++ procaddr.go	2018-04-08 19:28:35.356680000 +0200
@@ -15,9 +15,9 @@
 #cgo darwin CFLAGS: -DTAG_DARWIN
 #cgo darwin LDFLAGS: -framework OpenGL
 #cgo linux CFLAGS: -DTAG_LINUX
-#cgo linux LDFLAGS: -lGL
+#cgo linux,!egl freebsd,!egl pkg-config: gl
+#cgo linux,egl freebsd,egl pkg-config: egl
 #cgo egl CFLAGS: -DTAG_EGL
-#cgo egl LDFLAGS: -lEGL
 // Check the EGL tag first as it takes priority over the platform's default
 // configuration of WGL/GLX/CGL.
 #if defined(TAG_EGL)
```